### ISSUE TITLE
weekly checkin meeting 2018-01-09 11am EST

### ISSUE THREAD

**Issue by rabernat** [pangeo-data/pangeo#527]
Happy new year Pangeans! I hope your holidays were restful.

I am excited for our first weekly checkin meeting of the new year. In addition to the usual status updates, I think this would be a good opportunity to think strategically about what we want to accomplish in the next year. High on my list are the following two items:

- **The Pangeo Data Catalog** - I am increasingly convinced we need to move fast on prototyping an actual data catalog for all of the zarr assets we are putting into the catalog. There has been lots of discussion on this in recent (and old) issues (e.g. #503, #502, #507, https://github.com/raphaeldussin/example.pangeo.io-deploy/issues/9). I am also now reading up about [STAC](https://github.com/radiantearth/stac-spec) and am increasingly convinced it should be the foundation for such a catalog (see https://github.com/radiantearth/stac-spec/issues/366). I would like to have some discussion about these issues at the meeting and perhaps engage in some online whiteboarding (if such a thing is possible).

- **Pangeo for Machine Learning** - Many of us agree that Pangeo is a perfect platform for large-scale machine learning / deep learning (see e.g. #523). But there are some issues that need to be addressed in order to realize this potential (e.g. streaming data from xarray / zarr to DL libraries, provisioning GPUs, etc.). I would like to get peoples' thoughts on this as well. Pinging @nbren12, @RPrudden, and @alando46, who have expressed interest in the topic.

 Looking forward to re-connecting with everyone.

https://appear.in/pangeo

**Comment by ktyle** [comment]
Would definitely echo Ryan A.'s points for machine learning. I'm attending a ML short course at AMS presently. Chatted briefly with Ryan Lagerquist of OU/CIMMS who was not aware of Pangeo. Would be good to entrain him and also David John Gagne, one of the other course leaders.

**Comment by rabernat** [comment]
Glad folks are interested in this. I will shamelessly plug the very, very immature library I have started called xbatcher: https://xbatcher.readthedocs.io/en/latest/, which is designed to interface xarray datasets with keras [fit_generator](https://keras.io/models/model/#fit_generator).

**Comment by jhamman** [comment]
Thanks for organizing @rabernat. At AGU, we talked about a Pangeo Machine Learning Working Group. I'll try to bring this up more in our call on Wednesday.

**Comment by RPrudden** [comment]
On streaming data, one option might be to point a PyTorch [Dataset](https://github.com/pytorch/pytorch/blob/master/torch/utils/data/dataset.py) at the preferred kind of lazy array (xarray, etc.). Then we could let other PyTorch classes like [DataLoader](https://github.com/pytorch/pytorch/blob/master/torch/utils/data/dataloader.py) or [DistributedSampler](https://pytorch.org/docs/stable/_modules/torch/utils/data/distributed.html) handle the rest of the loading.

**Comment by alando46** [comment]
Reproducibility is one of the motivations behind the Pangeo project that resonates most with me. In addition to discussion around the mechanical components of distributing training/prediction, we should probably think about standards/tools/workflows that could help automate version controlling of training data, pipelines and models (data provenance).

[Pachyderm](https://pachyderm.io/open_source.html) is one (partially) open source example of how to do this. I imagine members of the Pangeo community have other examples/ideas for how to accomplish data/pipeline/model versioning.

Seems like there might be a lot to discuss. If it makes sense to focus on just STAC or just the technical aspects of ML systems for this call (or some subset of all the things that could be included), I'm in full support. Either way, looking forward to the call.

**Comment by nbren12** [comment]
I would be happy to join the conversation (particularly about the ML stuff)! @raspstephan might also be interested.  

FWIW, I have been using exactly the approach that @RPrudden mentions for training my PyTorch models with xarray data. Here is my [implementation](https://github.com/nbren12/uwnet/blob/master/uwnet/datasets.py). Data loading is the main bottleneck in my training procedure, so I would be thrilled if pangeo could solve that for me. A big challenge is that stochastic gradient descent accesses data at random indices, so typical chunking strategies will be very slow.

**Comment by nbren12** [comment]
> I am generally of the opinion that it makes sense to save your data to disk again for training deep learning models.

I completely agree with you Stephan, I am pretty sure this what @stephanrasp does. I am not familiar with TFRecord files, but I am not sure PyTorch has an equivalent.

As an aside, I am sure we can learn something from the approaches the deep learning team from Lawrence Berkeley Lab used to earn their [Gordon Bell Prize](http://cs.lbl.gov/news-media/news/2018/lawrence-berkeley-oak-ridge-national-labs-share-2018-acm-gordon-bell-prize/). The really pushed the envelope for deep learning of climate data on HPC.

**Comment by arokem** [comment]
I will try to join tomorrow's meeting to listen in and make a habit of it. I plan to resume my work on neuro.pangeo (#399) in the next couple of weeks, and can probably report on progress in a few weeks. 

As an aside, I noticed that the calendar here: http://pangeo.io/meeting-notes.html, is showing incorrect information (either that or the title of this issue is incorrect...).

**Comment by jhamman** [comment]
Sorry for just noticing this now. The call tomorrow should be scheduled for 3p ET tomorrow based on our current early/late rotation: http://pangeo.io/meeting-notes.html

I personally available at both times but I fear we have some crossed wires here...

**Comment by rabernat** [comment]
Oops my bad! How should we resolve this? I'm inclined to stick with the time proposed in this issue, and, in fact, I am not really available at 3pm ET (9pm eurotime). 

Let's take a poll here:
- Give a 👍 to this comment if you want to **stick with 11am EST**
- Give a 👎 to this comment if you want to **move to 3pm EST**

**Comment by niallrobinson** [comment]
Hi all, we've got a VIP today so the Met Office won't be able to make the call. I strongly endorse @rabernat s two initial points as priorities.

I also think that version control/publishing are high up the list after those things. We're really interested in https://medium.com/netflix-techblog/notebook-innovation-591ee3221233 and the potential of this kind of workflow for our use case, and have been speaking to these guys https://dotscience.com/ about how we might be able to adapt their system to Pangeo

**Comment by shoyer** [comment]
Sorry I'm missing the call this morning. I'd be happy to follow up with anyone about Pangeo for ML use cases.
